### PR TITLE
Fix cyclomatic complexity in cmd/controller/main.go

### LINKED ISSUE #177: Fix cyclomatic complexity in cmd/controller/main.go
Remove the `// nolint` comment.

The main function has a cyclomatic complexity greater than 10. Refactor the function to lower it.

### PR DESCRIPTION
**Pull request by enocom**
Fixes #177.

### CODE REVIEW

**Comment by EricFortin** [comment]
I would have merged it but it says the branch needs to be updated. There is also a button offering to update, should I do it or let @enocom do it himself?

**Comment by markmandel** [comment]
Can we rebase this down to a single commit please?

**Comment by enocom** [comment]
I'll update the branch and squash all the commits.

File: cmd/controller/main.go [hunk 1]
```diff
@@ -101,95 +163,60 @@ func main() { // nolint: gocyclo
...
+		certFile:          certFile,
 	}
+}
 
-	kubeClient, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		logger.WithError(err).Fatal("Could not create the kubernetes clientset")
-	}
+// ctlConfig stores all required configuration to create a game server controller.
+type ctlConfig struct {
```
Comment (markmandel) [inline cmd/controller/main.go]: Super minor nit - should this just be `config` - it's not like we have any other configurations around? Happy if you ignore this, but thought I'd ask.

File: pkg/fleets/controller.go [hunk 2]
```diff
@@ -49,6 +49,7 @@ import (
 // Controller is a the GameServerSet controller
 type Controller struct {
 	logger              *logrus.Entry
+	threadiness         int
```
Comment (markmandel) [inline pkg/fleets/controller.go]: So looking at other controllers, such as [Kubernetes controllers](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller.go#L137) or the [sample controller](https://github.com/kubernetes/sample-controller/blob/master/controller.go#L156) they have a `Run()` signature of `Run(threadiness int, stopCh <-chan struct{})` (although some of the naming differs from place to place).

Should we be aligning with the Kubernetes convention here / are we losing anything by not doing that. Maybe we lose a bit of ability for people to transfer Controller building knowledge when moving to Agones? (although it's minimal)

This isn't something I'm _super_ worried about, but figured it was worth raising - and if we find it a real issue down the line, it's pretty easy to back out of.

WDYT?

Comment (enocom) [reply pkg/fleets/controller.go]: That's a good discovery. I'll adjust to match it.

File: cmd/controller/main.go (hunk 1 above)
Comment (enocom) [reply cmd/controller/main.go]: Sure. I'll rename it.

**Comment by markmandel** [comment]
LGTM - will hit approve & merge when it's rebased to a single commit :+1: 

Nice stuff!

**Comment by enocom** [comment]
Just squashed the commits.

**Pull request merged**